### PR TITLE
Adaptations for Panua Pardiso

### PR DESCRIPTION
**Pull request by j-fu**
Hi, trying to add support for Pardiso7 here. AFAIK, it has no breaking API changes wrt. 6.x. 

However runtests bailed out in the indefinite hermitian case as Pardiso doesn't report an error in this case. Nevetheless, it does not solve, so I compare hashes of X in order to catch this case. May be this is not the smartest way, but works.

Will discuss this with upstream.

### CODE REVIEW

**Comment by KristofferC** [comment]
> However runtests bailed out in the indefinite hermitian case as Pardiso doesn't report an error in this case. 

Is there no error code at all or something else that can be inspect for this?

**Comment by j-fu** [comment]
Just contacted upstream wrt to this. I assume it is a bug in Pardiso (I checked: Err[] is zero), as it _knows_ to do nothing in this case. EDIT: We may wait for a response...

**Comment by j-fu** [comment]
The PR hashes X before and after, and figures this out (so runtests works with on my system), but this is not optimal.

**Comment by j-fu** [comment]
Hi, I contacted upstream, contact was acknowledged, but I did not get a response so far, they must be really busy. I propose to move on with the PR and keep the hashing hack in order to get Pardiso7 functionality, the more that new users won't be able to get Pardiso6 anymore.
- IMHO the hashing hack is reasonably fast (anyway, issymmetric  is called etc.) and reliable
- Extrapolating from their release policy I would assume that fixes will be provided in a new release which may be linked with some other features.

Once the fix is in, the hashing hack is easily removed.

**Comment by j-fu** [comment]
Seems upstream dialed back the current version to Pardiso6, so we can keep this PR on hold so far.

**Comment by KristofferC** [comment]
> Seems upstream dialed back the current version to Pardiso6, 

What is upstream here?

**Comment by j-fu** [comment]
The guys at pardiso-project.org I am in contact with (Olaf Schenk). 

I the moment, licensing and  download at pardiso-project.org provide 6.0 instead of 7.2.

**Comment by KristofferC** [comment]
Alright!

**Comment by j-fu** [comment]
Pardiso8 is out, now  provided by a new startup, Panua. They now provide 12 month user and machine locked academic licenses and verify academic affiliation. The problem with the hermitian indefinite case persists, I contacted Olaf Schenk on this.

**Comment by mipals** [comment]
@j-fu Do you know if this code works for Pardiso8?

**Comment by j-fu** [comment]
I didn't test this yet, will try to find time.

**Comment by j-fu** [comment]
Ok got the version from panua: `libpanuapardiso-20230908-linux.so` .
Tests for Float64/Complex64 work for unsymmetric, hermitian posdef. 
Tests fail for hermitian indef and symmetric. Didn't check deeper.

I guess for installation now we would tell users: "rename libpanuapardiso-YYYYMMDD-linux.so to libpanuapardiso.so", or is there any way to pass parameters to "deps/build.jl" ? 
EDIT: Probably an environment variable...

**Comment by j-fu** [comment]
Panua changed license conditions:
```
Full academic licence
Unlimited number of cores.
Renewable.
Only for academic users.
Tied to a single user & node.
Duration: 365 days
Price: 300.00 CHF
```
```
Free academic licence
Limited to 2 cores.
Renewable.
Only for academic users.
Tied to a single user & node.
Duration: 30 days
Price: Free
Obtain academic trial licence
```

**Comment by j-fu** [comment]
```
Full academic cluster licence
Unlimited number of cores.
Unlimited number of nodes.
Only for academic users.
Limited to a single user & cluster.
Duration: 365 days
Price: 1200.00 CHF
```

**Comment by mipals** [comment]
Thanks a lot for checking it out Jürgen. A shame that the free license only covers two cores 😑

**Comment by mipals** [comment]
Hi again, now I've gotten Pardiso 8 to work on apple silicon. However, as you mentioned Jürgen the test fail for hermitian indef and symmetric. I think that there might be a problem with the try-catch for Hermitian matrices in the solve function.

**Comment by KristofferC** [comment]
I am not in academia anymore so I can't really get a license and test things out. If anyone wants to take over maintenance for this package write here and I will give you commit access.

**Comment by j-fu** [comment]
> I am not in academia anymore so I can't really get a license and test things out. If anyone wants to take over maintenance for this package write here and I will give you commit access.

... thinking about it ...  I am in academia, and also aquainted with two of the main authors of Pardiso.

**Comment by KristofferC** [comment]
I've invited you as a collaborator to the repo.

**Comment by mipals** [comment]
I will be in academia for at least one more year and would like to use this package meanwhile. Would be happy to help with the maintenance for now, but I am pretty inexperienced when it comes to contributing to open-source. So maybe it is enough with just Jürgen?

**Comment by KristofferC** [comment]
I added you as well so you can keep an eye on each other ;)

**Comment by mipals** [comment]
Haha, thanks @KristofferC , sounds like a good idea 👀 

@j-fu I saw in the email from Olaf that he was asking for some Julia code that reproduces the issue. Will you send him some? If not I would be happy to do it.

**Comment by j-fu** [comment]
I first wanted to rename and update this PR and try this out with Panua Pardiso (had it running already), as I ran out of pardiso8 license. If you have something ready immediately, please feel free to send it to Olaf.  It feels the issue is the same since the opening of this PR (though I had no time to look more deeply into this): pardiso returns without solving and without indicating an error if it doesn't recognize the matrix type.

**Comment by mipals** [comment]
I have a working license already. I will send Olaf an example.

I agree that that the current issue is most likely the same issue that was at as for 7.2, i.e. that Pardiso does not return an error when it should.

**Comment by rayegun** [comment]
Do we have any way to redistribute as a JLL as we do for MKL? I suspect not from a cursory look at their website. I am also getting an academic license and can help with maintenance.

**Comment by j-fu** [comment]
That of course would make things more robust, however there would still be the question of a license in sync with the version in the JLL. AFAIK, newly obtained licenses would not work with with older pardiso versions.

**Comment by rayegun** [comment]
Ah it looks like the free licenses expire within 30 days, I'm not sure they're going to agree to redistribution in light of that

**Comment by mipals** [comment]
Just a quick update: I've just helped Dimosthenis from Panua testing out a new binary. It seems like they have now fixed the  issue on their end i.e. Panua-Pardiso now return the correct error code.

**Comment by j-fu** [comment]
So they will provide an update of their download then ?

**Comment by mipals** [comment]
Yes @j-fu they will provide an updated binary on their website asap. They might need to do a few more tests, but I assume that it wont take long.

File: README.md [hunk 1]
```diff
@@ -31,19 +33,21 @@ If you rather use a self installed MKL follow these instructions:
   information.
 * Note that the `MKLROOT` environment variable must be set whenever using the library.
 
-### PARDISO 6.0
+### PARDISO from [panua.ch](https://pardiso-project.org) ("ProjectPardiso")
```
Comment (KristofferC) [inline README.md]: Should it just be called "Panua-Pardiso" everywhere instead of "project"?

Comment (j-fu) [reply README.md]: Yes, I think so. I wanted to go over this again when we can test the new version -- Mikkel @mipals on mac, me on Linux + Windows (I have a windows VM on my linux laptop...). I also  think about introducing an API call `panua_pardiso_available()`  wrapping `PARDISO_LOADED[]`.

**Comment by j-fu** [comment]
Ran PanuaPardiso tests sucessfully on 1.6, 1, nigthly, polished things a bit.

How do we release ? As 0.5.6 , 0.6 or 1.0 ? 

I prefer 1.0 as this makes semver more clear.

File: Project.toml [hunk 2]
```diff
@@ -4,15 +4,21 @@ repo = "https://github.com/JuliaSparse/Pardiso.jl.git"
...
+Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
 Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 MKL_jll = "856f044c-d86e-5d09-b602-aeab76dc8ba7"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 
 [compat]
+Compat = "4.10.0"
 MKL_jll = "2021, 2022, 2023, 2024"
+Libdl= "1.6"
```
Comment (KristofferC) [inline Project.toml]: Formatting

File: Project.toml [hunk 3]
```diff
@@ -4,15 +4,21 @@ repo = "https://github.com/JuliaSparse/Pardiso.jl.git"
...
 
 [deps]
+Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
 Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 MKL_jll = "856f044c-d86e-5d09-b602-aeab76dc8ba7"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 
 [compat]
+Compat = "4.10.0"
```
Comment (KristofferC) [inline Project.toml]: Can we just handroll this instead? I really dislike Compat :/.

Comment (j-fu) [reply Project.toml]: I underastand... We quite well just might export these as it is unlikely that they interfer with other names. Would be the easiest way IMHO.

Comment (j-fu) [reply Project.toml]: Done this now.

**Comment by j-fu** [comment]
Tested successfully by me on  linux, and by @mipals on mac.

I tried a windows test on my vm, get an error when loading the dll:
"The specified module could not be found". Perhaps I am missing some library (the pardiso manual states that Intel compilers and vscode need to be installed). 

I am lacking time to continue with the windows stuff. 

How do we proceed  ? Ask someone to have a test on windows ? @PetrKryslUCSD comes to my mind. AFAIK he is on windows and might be interested anyway.

**Comment by mipals** [comment]
I've only tested the new binary on master @j-fu . Has something changed with the loading of the binary on this branch? Should I try it out?

**Comment by j-fu** [comment]
Yes, please

**Comment by mipals** [comment]
Still fairly new to all of this. So I might be doing something wrong. But trying out this branch on its own does not run as it does not include the fixes for apple sillicon (as seen on master) - Meaning that I can not run the code due MKL always being loaded in.

**Comment by KristofferC** [comment]
> But trying out this branch on its own does not run as it does not include the fixes for apple sillicon (as seen on master)

It would have to be rebased on master to pick up those fixes. `git rebase origin/master` for example.

**Comment by mipals** [comment]
Ahh, it is there - It was just me who forgot to change to the j-fu-pardiso7 branch 😅 It seems to work on my end!

**Comment by j-fu** [comment]
The main diff now are the *_avaiable functions and the README. @mipals may be you have another look there. Guess this it will be ready to merge and to release.
I think it is non-breaking, so could be 0.5.6.

File: Project.toml [hunk 4]
```diff
@@ -13,6 +13,7 @@ SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 MKL_jll = "2021, 2022, 2023, 2024"
 julia = "1.6"
 
+
```
Comment (KristofferC) [inline Project.toml]:
```suggestion
```

**Comment by j-fu** [comment]
Will squash and merge in a couple of hours.

**Pull request merged**